src/main/java/io/vertx/core/datagram/impl/DatagramSocketImpl.java

```diff
@@ -210,16 +212,37 @@ public DatagramSocket resume() {
   @Override
   @SuppressWarnings("unchecked")
   public DatagramSocket send(Buffer packet, int port, String host, Handler<AsyncResult<DatagramSocket>> handler) {
+    Objects.requireNonNull(packet, "no null packet accepted");
     Objects.requireNonNull(host, "no null host accepted");
-    ChannelFuture future = channel().writeAndFlush(new DatagramPacket(packet.getByteBuf(), new InetSocketAddress(host, port)));
-    addListener(future, handler);
+    InetSocketAddress addr = InetSocketAddress.createUnresolved(host, port);
+    if (addr.isUnresolved()) {
```

Comment:
Ok. I assumed (wrongly) that a "host" which is actually just an IP address will always be "resolved" .

Need some easy way to check if the "host" needs resolution or not as don't want to do an async resolve on each send if the user has specified IP.